pkg/manager/internal.go

```diff
@@ -495,8 +503,22 @@ func (cm *controllerManager) startNonLeaderElectionRunnables() {
 			// we use %T here because we don't have a good stand-in for "name",
 			// and the full runnable might not serialize (mutexes, etc)
 			log.V(1).Info("non-leader-election runnable finished", "runnable type", fmt.Sprintf("%T", ctrl))
+			returnCh <- struct{}{}
 		}()
 	}
+
+	doneCount := 0
+
+	numRunners := len(cm.nonLeaderElectionRunnables)
+	for doneCount < numRunners {
+		select {
+		case <-returnCh:
+			doneCount++
+		default:
+		}
+	}
```

Comment:
This works and I'm fine with it, though it does make me think of a sync.WaitGroup, are we effectively reimplementing that?